### PR TITLE
feat: add CacheSetAddElement that accepts a single element

### PR DESCRIPTION
**Pull request by virratanasangpunth**
For integ-testing, since CacheSetAddElement simply delegates
to CacheSetAddElements, the integ test cases are basic smoke
test cases and are not as exhaustive as CacheSetAddElements
integ test cases.

https://github.com/momentohq/client-sdk-javascript/issues/170

### CODE REVIEW

File: src/simple-cache-client.ts [hunk 1]
```diff
@@ -165,6 +166,30 @@ export class SimpleCacheClient {
...
+  public async setAddElement(
+    cacheName: string,
+    setName: string,
+    element: string | Uint8Array,
+    ttl?: CollectionTtl
+  ): Promise<CacheSetAddElement.Response> {
+    return await this.setAddElements(
+      cacheName,
+      setName,
+      [element] as string[] | Uint8Array[],
```
Comment (virratanasangpunth) [inline src/simple-cache-client.ts:188]: uggggh this is an ugly casting that I can't figure out how to remove.

File: src/simple-cache-client.ts [hunk 2]
```diff
@@ -165,6 +166,32 @@ export class SimpleCacheClient {
...
+    ttl?: CollectionTtl
+  ): Promise<CacheSetAddElement.Response> {
+    return (
+      await this.setAddElements(
+        cacheName,
+        setName,
+        [element] as string[] | Uint8Array[],
+        ttl
+      )
+    ).toSingularResponse();
```
Comment (virratanasangpunth) [inline src/simple-cache-client.ts:192]: LMK if someone can figure out how to get rid of this ugly cast `as string[] | Uint8Array[]`

File: src/simple-cache-client.ts [hunk 3]
```diff
@@ -165,6 +166,32 @@ export class SimpleCacheClient {
...
+    cacheName: string,
+    setName: string,
+    element: string | Uint8Array,
+    ttl?: CollectionTtl
+  ): Promise<CacheSetAddElement.Response> {
+    return (
+      await this.setAddElements(
+        cacheName,
+        setName,
+        [element] as string[] | Uint8Array[],
```
Comment (virratanasangpunth) [inline src/simple-cache-client.ts:189]: LMK if someone can figure out how to get rid of this ugly cast as string[] | Uint8Array[]

Comment (cprice404) [reply src/simple-cache-client.ts:189]: you could just define the var as an explicitly typed const first:

```
const elements: string[] | Uint8Array[] = [element];
```

Comment (virratanasangpunth) [reply src/simple-cache-client.ts:189]: The compiler won't allow that... unless I do the same ugly cast:
```
const elements: string[] | Uint8Array[] = [element] as string[] | Uint8Array[]
```

Comment (schwern) [reply src/simple-cache-client.ts:189]: If it makes you feel better, I ran into the same problem and did something even worse.

```
      if (typeof props.value === 'string') {
        return Momento.listConcatenateFront(
          props.cacheName,
          props.listName,
          [props.value],
          props.ttl,
          props.truncateBackToSize
        );
      } else {
        return Momento.listConcatenateFront(
          props.cacheName,
          props.listName,
          [props.value],
          props.ttl,
          props.truncateBackToSize
        );
      }
```

**Review by schwern — DISMISSED** [review]

**Review by schwern — APPROVED** [review]

**Pull request merged**